File: 고득점 kit/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <stack>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    stack<int> st;
    
    for (int a : arr) {
        // 비어있으면 넣으셈
        if (st.empty()) {
            st.push(a);
            answer.push_back(a);
        }
        
        else {
            // 스택의 최상단과 같다면
            // 중복이므로 패스
            if (st.top() == a)
                continue;
            st.push(a);
            answer.push_back(a);
        }
    }

    return answer;
}

File: 고득점 kit/스택과큐/같은 숫자는 싫어.swift
import Foundation

func solution(_ arr: [Int]) -> [Int] {
    var answer:[Int] = [] // 정답 
    var stack:[Int] = [] // 스택

    // 단순풀이
    for i in 0..<arr.count {
        if stack.count == 0 {
            st.append(arr[i])
            answer.append(arr[i])
        } else { 
            // 같은 숫자를 보면
            if st.last == arr[i] {
                continue
            }
            // 다른 숫자이니 넣어줌
            st.append(arr[i])
            answer.append(arr[i])
        }
    }
    return answer
}